igor-web/src/main/groovy/com/netflix/spinnaker/igor/IgorConfigurationProperties.groovy

```diff
@@ -42,6 +42,19 @@ class IgorConfigurationProperties {
 
         @NestedConfigurationProperty
         final BuildProperties build = new BuildProperties()
+
+        @Canonical
+        static class PollingSafeguardProperties {
+            /**
+             * Defines the upper threshold for number of new cache items before a cache update cycle will be completely
+             * rejected. This is to protect against potentially re-indexing old caches due to downstream service errors.
+             * Once this threshold is tripped, API-driven resolution may be required.
+             */
+            int itemUpperThreshold = 10
```

Comment:
Aren't you worried that we can legitimately hit that limit on any given Jenkins master? Do we have metrics around the cache change rate as of today?

Reply:
What does API-driven resolution mean?

Reply:
everything is fine ... re-index w/o sending notifications